### DISCUSSION TITLE
Toggle password visibility feature

### DISCUSSION THREAD

**Discussion by sheriffderek** [formkit/formkit#696] [Ideas]
I'm curious if there are any paths in FormKit for password visibility toggles.

<img width="526" alt="CleanShot 2023-05-09 at 18 39 12@2x" src="https://github.com/formkit/formkit/assets/2066617/2dfffa80-a0ea-4434-aa79-628cf3d3805c">

<img width="529" alt="CleanShot 2023-05-09 at 18 39 17@2x" src="https://github.com/formkit/formkit/assets/2066617/41ee82aa-f284-412d-95dc-b8db285bd138">

There's already a spot in the markup where the appended icons can go, but I don't see any discussion about it. Here's a quick CodePen:  https://codepen.io/perpetual-education/pen/poxpObG

If there's not already one that I'm not seeing, then I'd like to request this. : )

I'd also like the added bonus of connecting a password and a password confirmation field so that the toggle is optionally toggled on both fields (when noted as connected).

I'll also look into the "Create your own field" story.

🔓 

<img width="418" alt="CleanShot 2023-05-09 at 18 35 20@2x" src="https://github.com/formkit/formkit/assets/2066617/cbf4df67-edfd-4338-8051-73e7e799e45d">

**Comment by andrew-boyd** [comment]
We actually cover this in the Icons documentation as one of the examples. You can find a functioning demo here: 
https://formkit.com/essentials/icons#icon-click-handlers

Binding the behavior across inputs would be as simple as setting the `type` and `suffix-icon` values to `refs` and then using them on both your password and password confirmation fields so that when one field is toggled it affects both inputs.

https://formkit.link/047c90345f40bbea88182d29ee38165b

**Comment by sheriffderek** [comment]
Excellent. Very cool. 

For some reason, I assumed those icons were just going to be static. But as it so often happens... I'm reminded that we can make pretty much _anything_ happen. Thank you!
